### PR TITLE
chore: Release 8.4.0

### PR DESCRIPTION
**Pull request by ShGKme**
## [v8.4.0](https://github.com/nextcloud-libraries/nextcloud-vue/tree/v8.4.0) (2023-12-22)
[Full Changelog](https://github.com/nextcloud-libraries/nextcloud-vue/compare/v8.4.0...v8.3.0)

### Notes
* NcRichContenteditable: new visual label feature might break layout if you have custom contenteditable styles
* `isMobile` and `isFullscreen` mixins are now deprecated, use `useIsMobile` and `useIsFullscreen` composables instead

### 🚀 Enhancements
* enh: Split translations by components to only include needed strings in app bundles by @susnux in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4861
* enh(NcCheckboxRadioSwitch): Allow to set `aria-label` by @susnux in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4903
* enh(NcRichContenteditable): Add optional visual label for content editable by @susnux in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4907
* enh(NcBreadcrumbs): Allow setting aria-label on nav by @Pytal in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4913
* feat: add small mobile breakpoint by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4919
* Add target prop for NcListItem by @Fenn-CS in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4923
* Make modal mask dark if some icons are shown outside by @szaimen in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4916
* enh: Add a full labelled color palette to be used as the default for the color picker by @susnux in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4902

### 🐛 Fixed bugs
* fix(NcAppNavigationItem): fix style when using `active` prop by @raimund-schluessler in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4989
* fix(NcDialog): Increase specificity for modal-content selector by @susnux in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4897
* Prevent NcBreadcrumb from automatically inheriting attributes by @JuliaKirschenheuter in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4918
* fix(NcAvatar): Increase contrast of avatar status icon by @Pytal in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4912
* fix(utils/UserStatus): typo in "invisible" translation key by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4931
* enh(NcDialog): Label dialog nav by @Pytal in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4928
* fix(NcAppNavigationSettings): remove incorrect aria-label by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4932
* fix(NcAppSidebar): add focus trap on mobile by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4909
* feat(NcRichContenteditable): put caret in the end when focus input by @Antreesy in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4924
* fix(package): specify exports for composables by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4951
* fix(NcActions): provide `aria-expanded="false"` when menu is closed instead of removing it by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4946
* fix(NcListItem): Ensure list item does not overflow wrapper by @susnux in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4956
* fix(NcActionX): Allow relative icon URL by @susnux in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4955
* fix(NcAppSettingsDialog): respect showNavigation prop by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4954
* fix(NcAppNavigation): Bring back hover state for active element by @susnux in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4960
* enh(aria): changed check color to adhere to 3:1 ratio by @emoral435 in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4958
* Use logical style for NcHeaderMenu by @ahangarha in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4965
* Equalize `TextArea` styles with `TextField` styles by @JuliaKirschenheuter in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4971
* Increase contrast on input field for `NcRichContenteditable` by @JuliaKirschenheuter in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4975
* enh(breadcrumbs): changed aria label for voice control on a-tag by @emoral435 in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4973
* Differentiate between dark and light theme for NcDateTimePickerNative by @JuliaKirschenheuter in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4981

### Other Changes
* refactor(NcAppSidebar): remove unneeded debug log by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4952
* chore(ci): make Jest output colorful by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4966
* Update translations

### CODE REVIEW

**Comment by ChristophWurst** [comment]
Is this the first release with full Vue 3 support? https://github.com/nextcloud-libraries/nextcloud-vue/pull/3692#issuecomment-1866505148

**Comment by ShGKme** [comment]
> Is this the first release with full Vue 3 support? [#3692 (comment)](https://github.com/nextcloud-libraries/nextcloud-vue/pull/3692#issuecomment-1866505148)

No, we should make `v9.0.0-alpha.1` release later from `next` branch.

**Comment by ShGKme** [comment]
But I forgot to mention deprecated stuff in the changelog.

**Comment by ahangarha** [comment]
Let me know if you think we should have logical CSS properties fully added in this release, and you can wait for one more week. Otherwise, they will be added in the next release.

**Review by AndyScherzinger — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,48 @@
...
+* fix(package): specify exports for composables by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4951
+* fix(NcActions): provide `aria-expanded="false"` when menu is closed instead of removing it by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4946
+* fix(NcListItem): Ensure list item does not overflow wrapper by @susnux in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4956
+* fix(NcActionX): Allow relative icon URL by @susnux in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4955
+* fix(NcAppSettingsDialog): respect showNavigation prop by @ShGKme in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4954
+* fix(NcAppNavigation): Bring back hover state for active element by @susnux in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4960
+* enh(aria): changed check color to adhere to 3:1 ratio by @emoral435 in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4958
+* Use logical style for NcHeaderMenu by @ahangarha in https://github.com/nextcloud-libraries/nextcloud-vue/pull/4965
```
Comment (ahangarha) [inline CHANGELOG.md:37]: I am not sure if this PR should be categorized under bug fixes. I see it more as adding new capability to the UI to adjust itself to the direction of its context.

**Review by ahangarha — CHANGES_REQUESTED** [review]
I have a proposition for changing the category of #4965

**Comment by ShGKme** [comment]
> Let me know if you think we should have logical CSS properties fully added in this release, and you can wait for one more week. Otherwise, they will be added in the next release.

We always can make a new library release when needed, so we can just make 8.4.1 release next week or in 2 weeks. Next Nextcloud server release is planned for 25 Jan 2024 anyway.

File: CHANGELOG.md (hunk 1 above)
Comment (susnux) [reply CHANGELOG.md:37]: I think it is a bug if it does not work for some languages?

Comment (ShGKme) [reply CHANGELOG.md:37]: I'd call it definitely a bug fix. The component didn't work correctly. A new feature is a new component/function or new feature of a component with a new property/event and ect.

**Review by susnux — APPROVED** [review]

**Comment by susnux** [comment]
But we can also wait for next year to get some more in as there will not much updating anyways

**Comment by ahangarha** [comment]
> Nextcloud server release is planned for 25 Jan 2024 anyway.

In any case, I would like to have all the logical style fixes before the NextCloud-server release so that it can use these styles. In any case, I should hurry up 🙂 🏃🏾‍♂️

**Review by JuliaKirschenheuter — APPROVED** [review]

**Review by raimund-schluessler — APPROVED** [review]

**Comment by ShGKme** [comment]
Rebased on `master` and to include one bug fix and great split translations enh.

**Pull request merged**